### ISSUE TITLE
Loki Will be using RandomXL  a custom RandomX variant 

### ISSUE THREAD

**Issue by SomethingGettingWrong** [xmrig/xmrig#1044]
testnet pool here: https://dev.imaginary.stream and a fork of xmrig that supports it here: https://github.com/jagerman/xmrig/tree/beta

Will you add to yours so that its integrated when OPENCL an CUDA Versions will be finished?

**Comment by trasherdk** [comment]
See #1028 :^)

**Comment by jagerman** [comment]
> See #1028 :^)

What does #1028 have to do with this?

**Comment by Spudz76** [comment]
RandomX already works on CPU and has for at least a week (if you compile from beta branch)
CUDA has both merged into dev branch a couple hours ago (again, compile)
OpenCL is apparently next but nothing to see yet.

All devices require 2080MB of VRAM to run RandomX - and that's before any work scratchpads - so 3GB or more VRAM is definitely a requirement.  Not sure about RandomXL but I think any RandomX-based-algo has the 2080MB requirement regardless of the scratchpad size or other tweaks.

So, if you expect to run RandomX on small VRAM cards (like CN) you will be quite literally unable.  This algo deprecates hardware almost as bad as dagger (at least the waste of VRAM doesn't grow with time?)

**Comment by SomethingGettingWrong** [comment]
Random X according to the designer has made it easy to modify it in config.h. **So there should be an expectancy of many variations in the coming months**.  I was letting the developer know so that he can go ahead and add Random XL into his miner. The links i posted in the original post were from a an fork for Loki project by one of the core Loki devs.

**Comment by jagerman** [comment]
@SomethingGettingWrong - the `evo` branch now has generic RandomX support, including for RandomXL, that supports RandomX configuration at run-time rather than compile-time.  (The current `beta` branch, though, only supports a linking to a single randomx lib; the change in `evo` fixes the issue.)

**Comment by Spudz76** [comment]
Aha!  I had switched to evo recently and thus did not know dev or beta didn't have this too.

**Comment by Spudz76** [comment]
Also the designer of the algo *and* the one doing the code here, are the same person.  I'm pretty sure they know how to use their own libs. :)
